### PR TITLE
Add report generation for method uses

### PR DESCRIPTION
**Pull request by romeara**

### CODE REVIEW

File: docs/REPORT_FORMAT.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+The canonical use for the library results in an output report. This report is a zip archive with a custom file extension, containing multiple report files. These files are partitioned groups of discovered methods (the number per partition is determined by cloud service constraints). There are currently two types of files - "methods", and "method uses".
+
+## Methods Report
+
+The "methods" report(s) are within the `referenced-methods` directory of the expanded report file. Each of these files is a JSON file, containing one element: `methodIds`. This is a JSON array of strings. Each string is a unique, opaque ID refering to a specific discovered method, and is contained as an identifier within the other report formats for linking data together.
+
+## Method Uses Report
+
+The "method uses" report(s) are within the `referenced-method-uses` directory of the expanded report file. Each of this files is a JSON file, containing several elements:
```
Comment (nickavv) [inline docs/REPORT_FORMAT.md]: "each of _these_" rather than "each of this"

Comment (romeara) [reply docs/REPORT_FORMAT.md]: Good catch - fixed!

**Comment by WyattPearson** [comment]
It looks like the metadata.json is now going to be omitted, is the expectation that that data is provided by the caller in other parts of the request?

**Comment by romeara** [comment]
@WyattPearson Appears I was working off an old copy of the POC, before meta-data was added as a file. I will address that in a follow-up PR

**Pull request merged**